src/data/matrix/basic.lean

```diff
@@ -644,6 +644,14 @@ lemma smul_mul_vec_assoc (A : matrix m n α) (b : n → α) (a : α) :
   (a • A).mul_vec b = a • (A.mul_vec b) :=
 by { ext, apply smul_dot_product }
 
+lemma mul_vec_add_distrib (A : matrix m n α) (x y : n → α) :
+  A.mul_vec (x + y) = A.mul_vec x + A.mul_vec y :=
+by { ext, simp only [mul_vec, pi.add_apply, dot_product_add] }
+
+lemma vec_mul_add_distrib (A : matrix m n α) (x y : m → α) :
+  vec_mul (x + y) A = vec_mul x A + vec_mul y A :=
+by { ext, simp only [vec_mul, pi.add_apply, add_dot_product] }
```

Comment:
You can shorten both these proofs as follows:
```suggestion
lemma mul_vec_add_distrib (A : matrix m n α) (x y : n → α) :
  A.mul_vec (x + y) = A.mul_vec x + A.mul_vec y :=
by { ext, apply dot_product_add }

lemma vec_mul_add_distrib (A : matrix m n α) (x y : m → α) :
  vec_mul (x + y) A = vec_mul x A + vec_mul y A :=
by { ext, apply add_dot_product }
```

Reply:
Although I think better names for these lemmas might be `mul_vec_add` and `add_mul_vec`, to match `mul_add` and `add_mul`?